### PR TITLE
abseil: add version 20230125.0

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **abseil/20230125.0**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/abseil/all/conanfile.py [hunk 1]
```diff
@@ -30,35 +30,54 @@ class AbseilConan(ConanFile):
...
+    def _compilers_minimum_version(self):
+        if self._min_cppstd == 11:
+            return {}
+        return {
+            "gcc": "6",
+            "clang": "5",
+            "apple-clang": "10",
+            "Visual Studio": "15",
+            "msvc": "191",
+        }
```
Comment (SpaceIm) [inline recipes/abseil/all/conanfile.py:49]:
```suggestion
        return "11" if Version(self.version) < "20230125.0" else "14"

    @property
    def _compilers_minimum_version(self):
        return {
            "14": {
                "gcc": "6",
                "clang": "5",
                "apple-clang": "10",
                "Visual Studio": "15",
                "msvc": "191",
            },
        }.get(self._min_cppstd, {})
```

I think it's more elegant.

File: recipes/abseil/all/conanfile.py [hunk 2]
```diff
@@ -30,35 +30,54 @@ class AbseilConan(ConanFile):
...
-            check_min_cppstd(self, 11)
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
         if self.info.options.shared and is_msvc(self):
```
Comment (SpaceIm) [inline recipes/abseil/all/conanfile.py:72]:
```suggestion
        if self.options.shared and is_msvc(self):
```

File: recipes/abseil/all/test_v1_package/CMakeLists.txt [hunk 3]
```diff
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 3.8)
```
Comment (SpaceIm) [inline recipes/abseil/all/test_v1_package/CMakeLists.txt:1]: The all content of this file can be replaced by:

```cmake
cmake_minimum_required(VERSION 3.1)
project(test_package)

include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
conan_basic_setup(TARGETS)

add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/../test_package
                 ${CMAKE_CURRENT_BINARY_DIR}/test_package)
```

File: recipes/abseil/all/conanfile.py (hunk 1 above)
Comment (Hopobcn) [reply recipes/abseil/all/conanfile.py:49]: What is the policy for setting these minimum compiler versions? They must be provided by the project itself (folly) or is something that can be freely set by CCI? Wouln't it be good to set minimum defaults for C++11 (or use `conan.tools.build.supported_cppstd`)?

Comment (SpaceIm) [reply recipes/abseil/all/conanfile.py:49]: Projects usually don't know the full range of supported compilers. They test few versions. Moreover for a given compiler, the range between "0 support of C++xx" and " full support of C++××" can be huge. Many projects using common C++11 features can be built by gcc4.9 while it has not full C++11 support.

File: recipes/abseil/all/test_v1_package/CMakeLists.txt (hunk 3 above)
Comment (toge) [reply recipes/abseil/all/test_v1_package/CMakeLists.txt:1]: @SpaceIm 
Thanks your views!
Fixed.

There are one difference between test_package and test_v1_package.
To replace test_v1_package, I have to modify CMakeLists.txt in test_package followings:

```
if(TARGET abseil::abseil)
target_link_libraries(${PROJECT_NAME}_global PRIVATE abseil::abseil)
else()
target_link_libraries(${PROJECT_NAME}_global PRIVATE CONAN_PKG::abseil)
endif()
```

Comment (SpaceIm) [reply recipes/abseil/all/test_v1_package/CMakeLists.txt]: Why? Just use abseil::abseil for the global target. It should be defined in both cases.

Comment (toge) [reply recipes/abseil/all/test_v1_package/CMakeLists.txt]: @SpaceIm 
Thanks!
I use absl::absl instead of CONAN_PKG::abseil.

**Review by uilianries — APPROVED** [review]
LGTM

**Review by StellaSmith — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Pull request merged**